contracts/superswap/RexSuperSwap.sol

```diff
@@ -0,0 +1,147 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+pragma abicoder v2;
+
+import { ISuperToken } from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperfluid.sol";
+import "./TransferHelper.sol";
+import "./interfaces/ISwapRouter02.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../ISETHCustom.sol";
+
+import "hardhat/console.sol";
+
+
+contract RexSuperSwap {
+  using SafeERC20 for ERC20;
+  ISwapRouter02 public immutable swapRouter;
+
+  address public constant MATICX = 0x3aD736904E9e65189c3000c7DD2c8AC8bB7cD4e3;
```

Comment:
Can we add this in the constructor so we don't have issues deploying to other network.

Reply:
Maybe something like a `superNativeToken` var, a while back I was working on it here - https://github.com/Ricochet-Exchange/ricochet-super-swap/pull/2

Reply:
Alright good point I added the magic address to the contract constructor as you suggested. Can you think of any other tests we should run on this swap or anything that could be missing?